homeassistant/components/tplink/sensor.py

```diff
@@ -164,7 +164,7 @@ def __init__(
         self.entity_description = description
         self._attr_unique_id = f"{legacy_device_id(device)}_{description.key}"
         if has_parent:
-            assert device.alias
+            assert isinstance(device.alias, str)
```

Comment:
Honestly, this would then name the device `""`, I am not sure if that's what we want. If the name can be falsey, we should take a default string to name the device (not sure if there is a recognizable name other than the alias)

Reply:
Yeah, that was my concern, too. So basically what we want is something like this, but I'm unsure if it's currently doable?
```
alias = device.alias or tr("Socket")
```
Having non-recognizable names when no name is given shouldn't be an issue given they can be renamed manually. Alternatively, we could use the device id (which is done for the unique_id) if no name is available, but they look rather ugly as they are 42-char hex-encoded strings.